starter-core/src/main/java/io/micronaut/starter/build/dependencies/PomDependencyVersionResolver.java

```diff
@@ -29,78 +28,79 @@
 import javax.xml.parsers.ParserConfigurationException;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Collectors;
 
 @Singleton
 public class PomDependencyVersionResolver implements CoordinateResolver {
 
     private static final String NODE_NAME_TEXT = "#text";
-    private final Map<String, Coordinate> coordinates;
 
-    public PomDependencyVersionResolver(ResourceResolver resourceResolver) {
+    /**
+     * Initialize coordinates early to remove runtime dependencies on javax.xml.
+     */
+    private static final Map<String, Coordinate> COORDINATES = readCoordinates();
+
+    private static Map<String, Coordinate> readCoordinates() {
         Map<String, Coordinate> coordinates = new HashMap<>();
-        for (URL url : resourceResolver.getResources("classpath:pom.xml").collect(Collectors.toList())) {
```

Comment:
I think you PR removes the ability to read multiple `pom.xml`. E.g. Guides has an extra `pom.xml` https://github.com/micronaut-projects/micronaut-guides/blob/master/buildSrc/src/main/resources/pom.xml

Reply:
Thank you for the tip @sdelamo . Do we know statically all the `pom.xml` files? Can we hardcode them, or is it possible to get hold of a resource resolver and use the code above?

Reply:
Can you change:

```java
private static final Map<String, Coordinate> COORDINATES = readCoordinates();

private static Map<String, Coordinate> readCoordinates() {
```

to: 

```java
private static final Map<String, Coordinate> COORDINATES = 
    readCoordinates(Collections.singletonList(Thread.currentThread()
    .getContextClassLoader()
    .getResourceAsStream("pom.xml")));

private static Map<String, Coordinate> readCoordinates(List<InputStream> pomInputStreams) {
```

I could override this bean in Guides to keep the previous behaviour.